dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/AbstractCommandExecutor.java

```diff
@@ -190,13 +191,13 @@ public TaskResponse run(IShellInterceptorBuilder iShellInterceptorBuilder,
             result.setExitStatusCode(this.process.exitValue());
 
         } else {
-            log.error("process has failure, the task timeout configuration value is:{}, ready to kill ...",
-                    taskRequest.getTaskTimeout());
+            log.error("process has failure due to timeout kill, timeout value is:{}, timeoutStrategy is:{}",
+                    taskRequest.getTaskTimeout(), taskRequest.getTaskTimeoutStrategy());
             result.setExitStatusCode(EXIT_CODE_FAILURE);
-            cancelApplication();
         }
         int exitCode = this.process.exitValue();
-        String exitLogMessage = EXIT_CODE_KILL == exitCode ? "process has killed." : "process has exited.";
+        String exitLogMessage = (EXIT_CODE_KILL == exitCode || EXIT_CODE_HARD_KILL == exitCode) ? "process has killed."
+                : "process has exited.";
```

Comment:
It's better to check from status in `taskExecutionContext` rather then check from the exit code, we don't care about the exit, only kill from ds then we think it's kill.

Reply:
OK